addons/mail/static/src/core/web/chatter.js

```diff
@@ -438,4 +472,24 @@ export class Chatter extends Component {
         }
         this.recipientsPopover.open(ev.target, { thread: this.state.thread });
     }
+
+    /**
+     * @param {string} searchTerm
+     */
+    onSearch(searchTerm) {
+        if (searchTerm) {
+            this.state.searchTerm = searchTerm;
+            this.threadService.search(searchTerm, this.state.thread).then((messages) => {
+                this.state.searchMessages = messages;
+            });
+        } else {
+            this.state.searchMessages = [];
+            this.state.searchTerm = undefined;
+        }
```

Comment:
This code is the exact same as in `search_messages_panel.js`. So can be moved as either a concern of `useMessageSearch` hook or `SearchForm`.

I suppose you wanted to keep `SearchForm` generic for showing input of search. So this can be in the component hook `useMessageSearch()` hook.